src/main/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java

```diff
@@ -129,45 +160,36 @@ public static void serialize(final IFactoryController controller, final Friendly
      */
     public static Tool deserialize(final IFactoryController controller, final FriendlyByteBuf buffer)
     {
-        final IToolType type = ToolType.getToolType(buffer.readUtf(32767));
+        ResourceLocation resLoc = ToolTypeEntry.parseResourceLocation(buffer.readResourceLocation());
+        final ToolTypeEntry type = ModToolTypes.getRegistry().getValue(resLoc);
```

Comment:
You do not have to do this for network buffer deserialization, only for NBT deserialization.
NBT is permanent storage on disk, network buffers can only be used during gameplay, so the data is already corrected on NBT load and can't be wrong right here.